layers/+lang/bibtex/packages-config.el

```diff
@@ -0,0 +1,57 @@
+;;; packages-config.el --- BibTeX Layer packages file for Spacemacs
+;;
+;; Copyright (c) 2012-2014 Sylvain Benner
+;; Copyright (c) 2014-2015 Sylvain Benner & Contributors
+;;
+;; Author: Joshua Ellis <josh@jpellis.me>
+;; URL: https://github.com/JP-Ellis
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Commentary:
+
+;;; Code:
+
+(defun bibtex/init-org-ref ()
+  (use-package org-ref
+    :defer t
+    :config
+    (progn
+      (spacemacs/set-leader-keys-for-major-mode 'bibtex-mode
+        ;; Navigation
+        "j" 'org-ref-bibtex-next-entry
+        "k" 'org-ref-bibtex-previous-entry
+
+        ;; Open
+        "b" 'org-ref-open-in-browser
+        "n" 'org-ref-open-bibtex-notes
+        "p" 'org-ref-open-bibtex-pdf
+
+        ;; Misc
+        "h" 'org-ref-bibtex-hydra/body
+        "i" 'org-ref-bibtex-hydra/org-ref-bibtex-new-entry/body-and-exit
+        "s" 'org-ref-sort-bibtex-entry
+
+        ;; Lookup utilities
+        "la" 'arxiv-add-bibtex-entry
+        "lA" 'arxiv-get-pdf-add-bibtex-entry
+        "ld" 'doi-utils-add-bibtex-entry-from-doi
+        "li" 'isbn-to-bibtex
+        "lp" 'pubmed-insert-bibtex-from-pmid
+        ))))
+
+(defun bibtex/post-init-auctex ()
+  (spacemacs/set-leader-keys-for-major-mode 'latex-mode
+    "ic" 'org-ref-helm-insert-cite-link))
+
+(defun bibtex/post-init-org ()
+  (spacemacs/set-leader-keys-for-major-mode 'org-mode
+    "ic" 'org-ref-helm-insert-cite-link))
+
+(defun bibtex/post-init-markdown-mode ()
+  (spacemacs/set-leader-keys-for-major-mode 'markdown-mode
+    "ic" 'org-ref-helm-insert-cite-link))
+
```

Comment:
At this stage, it's only three modes which need this, so I thought that repeating the code is not too bad, especially given that it's one command for each mode.  If, in the future, many more modes need a shortcut added then we can always use a DRYer way.

Having said that, if you want to do a pull request against my fork then I'll include it (though I'll probably squash it so as to have only 1 commit).